js/src/common/states/AlertManagerState.js

```diff
@@ -0,0 +1,40 @@
+export default class AlertManagerState {
+  constructor() {
+    this.activeAlerts = {};
+    this.alertId = 0;
+  }
+
+  getActiveAlerts() {
+    return this.activeAlerts;
+  }
+
+  /**
+   * Show an Alert in the alerts area.
+   */
+  show(attrs, type) {
```

Comment:
May I suggest renaming `type` with `component` or even `componentClass` ?

I find it confusing to have both a parameter and an attr/prop named `type`. People might confuse the two.

Reply:
People only supply the value for the argument, though - argument names aren't part of the public API.

Are you suggesting to change the property of the alert POJO as well? If not, I don't mind changing the parameter name back - something like `componentClass` was the original proposal anyway, IIRC.

Reply:
I'd say both the argument and the POJO attribute to reduce confusion to the maximum.

The attribute of the POJO should indeed never be visible to the developers but for our own sanity I'd argue it would make sense to call it `componentClass` so there's no ambiguity when reviewing the code.

The name of the argument might not be part of the public API per say but it might be visible in IDEs that show autocompletion and documentation, even more so once we have proper typescript definitions shipped.

Reply:
I don't care enough, whatever is clearer to you guys. :wink: